### DISCUSSION TITLE
Cloudshell Performance Optimization Guidelines

### DISCUSSION THREAD

**Discussion by Penguinwizzard** [Azure/CloudShell#480] [General]
My team just moved to using Cloud Shell a lot more, and our biggest question at the moment is how to get the best performance out of the environment. Are there any published guidelines around what's fast and what isn't in Cloud Shell?

In case it helps provide an example of a question to answer, here's my team's scenario:

> We have per-user Azure Files accounts, each with one share for that user's cloud shells. To enable us to build and run all of our team's code, we provision dotnet sdks for dotnet 6, 7, and 8 into the storage account, as well as pre-populated checkouts of our DRI tooling repos. Doing a `dotnet run` using one of these dotnet SDKs for code also on the fileshare can take a minute or more to start running. Is there something we're missing, or is this inherent to the way that the attached storage works?

**Comment by mbifeld** [comment]
Hi @Penguinwizzard , thank you for the feedback.  The only idea I have is if the storage accounts are in secondary regions, meaning a different region than where Cloud Shell provisions containers, there would be some delay.  Could you check if your storage accounts are in primary or secondary regions?

**Primary regions are:** Southeast Asia, East US, West US, South Central US, North Europe, Central India, West Europe

**Secondary regions are:** Australia Southeast, Korea Central, Korea South, Australia East, Australia Central, Japan West, Japan East, East Asia, Canada East, Brazil South, West US 2, West US 3, Canada Central, North Central US, West Central US, Germany West Central, Switzerland North, Norway East, Jio India West, UK West, France Central, South India, West India, South Africa North, Sweden Central, UAE North, Qatar Central, Mexico Central, Spain Central
